### PR TITLE
[Fix] Accomodating the infra change for multidut ECN and PFCWD cases

### PR DESCRIPTION
**Pull request by selldinesh**
### Description of PR

Summary: Accomodating the infra change for multidut ECN and PFCWD cases
Fixes # (issue)
https://github.com/sonic-net/sonic-mgmt/issues/13389
https://github.com/sonic-net/sonic-mgmt/issues/13769

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?
To accomodate the infra change from PR 14127 
#### How did you do it?
Added a pytest fixture called get_snappi_ports and get_snappi_ports_for_rdma whcih selects the ports from the information provided in MULTIDUT_PORT_INFO in variables.py

### CODE REVIEW

File: tests/common/snappi_tests/snappi_fixtures.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def snappi_api_serv_port(duthosts, rand_one_dut_hostname):
     """
     duthost = duthosts[rand_one_dut_hostname]
     return (duthost.host.options['variable_manager'].
-            _hostvars[duthost.hostname]['snappi_api_server']['rest_port'])
+            _hostvars[duthost.hostname]['secret_group_vars']['snappi_api_server']['rest_port'])
```
Comment (sdszhang) [inline tests/common/snappi_tests/snappi_fixtures.py:48]: try  ansible/group_vars/<inv>/creds.yml, then this change may not be needed.

**Comment by sdszhang** [comment]
@rraghav-cisco  @amitpawar12 can you try this diff and see if any comments? Thanks.

File: tests/snappi_tests/multidut/ecn/test_multidut_dequeue_ecn_with_snappi.py [hunk 2]
```diff
@@ -39,48 +37,50 @@ def test_dequeue_ecn(request,
...
-    snappi_ports = get_multidut_tgen_peer_port_set(line_card_choice, snappi_port_list, config_set, 2)
-
-    testbed_config, port_config_list, snappi_ports = snappi_dut_base_config(dut_list,
-                                                                            snappi_ports,
-                                                                            snappi_api)
-
-    _, lossless_prio = rand_one_dut_lossless_prio.split('|')
-    skip_ecn_tests(duthost1)
-    skip_ecn_tests(duthost2)
+    for testbed_subtype, rdma_ports in multidut_port_info.items():
```
Comment (sdszhang) [inline tests/snappi_tests/multidut/ecn/test_multidut_dequeue_ecn_with_snappi.py:53]: In this for loop the testbed_config, port_config will be calculated multiple times. seems only the result in last iteration will be used for the run_ecn_test?

Comment (selldinesh) [reply tests/snappi_tests/multidut/ecn/test_multidut_dequeue_ecn_with_snappi.py:53]: this wont impact

File: tests/common/snappi_tests/snappi_fixtures.py [hunk 3]
```diff
@@ -614,52 +614,6 @@ def snappi_dut_base_config(duthost_list,
     return config, port_config_list, new_snappi_ports
 
 
-@pytest.fixture(scope="function")
-def get_multidut_snappi_ports(duthosts, conn_graph_facts, fanout_graph_facts):            # noqa: F811
```
Comment (sdszhang) [inline tests/common/snappi_tests/snappi_fixtures.py:618]: let's seperate the clean up in a different PR.

**Review by yejianquan — APPROVED** [review]
LGTM

**Pull request merged**